### ISSUE TITLE
Bandgap Prediction using MACE

### ISSUE THREAD

**Issue by amannsonii3** [ACEsuit/mace#910]
**Detailed Description**
Bandgap, unlike energy, is an intrinsic property of the system, unchanging with the size of the system (e.g. constructing supercells of the system, unlike energy, would yield the same constant bandgap). 
I was wondering if we were to replicate certain energy functions in MACE, but using mean or scatter_mean instead of sum or scatter_sum, we should be able to train a model that can predict bandgaps.

**Features Request**
New functions in MACECalculator for bandgap calculator and replication of certain energy functions in MACE modified for bandgaps.

**Modifications I tried**
I tried editing def forward within MACE and ScaleShiftMACE class in the models.py file. I also tried modifying the weighted_mean_squared_error_energy function in loss.py 

1. mace/modules/loss.py: 
```
return torch.mean(
        configs_weight
        * configs_energy_weight
        * torch.square((ref["energy"] - pred["energy"])) # Does it matter if we keep the division by num_atoms or not?
    )
```
2. mace/modules/models.py
```
e0 = scatter_mean(
            src=node_e0, index=data["batch"], dim=-1, dim_size=num_graphs
        )
```
```
energy = scatter_mean(
                src=node_energies, index=data["batch"], dim=-1, dim_size=num_graphs
            )  # [n_graphs,]
```

``` 
total_energy = torch.mean(contributions, dim=-1) # [n_graphs,] 
```

**Additional context**
P.S. I am not very comfortable with the MACE architecture like the developers and contributors. It would be great to have assistance in building this feature. Please let me know if such a feature already exists and I might have missed it
